zitadel/v2/application_oidc/funcs.go

```diff
@@ -53,81 +53,95 @@ func update(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
-			GrantTypes:               grantTypes,
-			AppType:                  app.OIDCAppType(app.OIDCAppType_value[appType]),
-			AuthMethodType:           app.OIDCAuthMethodType(app.OIDCAuthMethodType_value[authMethodType]),
-			PostLogoutRedirectUris:   postLogoutRedirectURIs,
-			DevMode:                  devMode,
-			AccessTokenType:          app.OIDCTokenType(app.OIDCTokenType_value[accessTokenType]),
-			AccessTokenRoleAssertion: accessTokenRoleAssertion,
-			IdTokenRoleAssertion:     idTokenRoleAssertion,
-			IdTokenUserinfoAssertion: idTokenUserinfoAssertion,
-			ClockSkew:                clockSkew,
-			AdditionalOrigins:        additionalOrigins,
-		})
+	if d.HasChanges(redirectURIsVar,
+		appTypeVar,
+		authMethodTypeVar,
+		postLogoutRedirectURIsVar,
+		devModeVar,
+		accessTokenTypeVar,
+		accessTokenRoleAssertionVar,
+		idTokenRoleAssertionVar,
+		idTokenUserinfoAssertionVar,
+		clockSkewVar,
+		additionalOriginsVar,
+	) {
+		respTypes := make([]app.OIDCResponseType, 0)
+		for _, respType := range d.Get(responseTypesVar).([]interface{}) {
+			respTypes = append(respTypes, app.OIDCResponseType(app.OIDCResponseType_value[respType.(string)]))
+		}
+		grantTypes := make([]app.OIDCGrantType, 0)
+		for _, grantType := range d.Get(grantTypesVar).([]interface{}) {
+			grantTypes = append(grantTypes, app.OIDCGrantType(app.OIDCGrantType_value[grantType.(string)]))
+		}
+		dur, err := time.ParseDuration(d.Get(clockSkewVar).(string))
 		if err != nil {
-			return diag.Errorf("failed to update applicationOIDC: %v", err)
+			return diag.FromErr(err)
+		}
+
+		oidcConfig := oidcApp.GetOidcConfig()
+		redirecURIs := interfaceToStringSlice(d.Get(redirectURIsVar))
+		appType := d.Get(appTypeVar).(string)
+		authMethodType := d.Get(authMethodTypeVar).(string)
+		postLogoutRedirectURIs := interfaceToStringSlice(d.Get(postLogoutRedirectURIsVar))
+		devMode := d.Get(devModeVar).(bool)
+		accessTokenType := d.Get(accessTokenTypeVar).(string)
+		accessTokenRoleAssertion := d.Get(accessTokenRoleAssertionVar).(bool)
+		idTokenRoleAssertion := d.Get(idTokenRoleAssertionVar).(bool)
+		idTokenUserinfoAssertion := d.Get(idTokenUserinfoAssertionVar).(bool)
+		clockSkew := durationpb.New(dur)
+		additionalOrigins := interfaceToStringSlice(d.Get(additionalOriginsVar))
+
+		if !reflect.DeepEqual(redirecURIs, oidcConfig.GetRedirectUris()) ||
+			!reflect.DeepEqual(respTypes, oidcConfig.GetResponseTypes()) ||
+			!reflect.DeepEqual(grantTypes, oidcConfig.GetGrantTypes()) ||
+			appType != oidcConfig.AppType.String() ||
+			authMethodType != oidcConfig.AuthMethodType.String() ||
+			!reflect.DeepEqual(postLogoutRedirectURIs, oidcConfig.GetPostLogoutRedirectUris()) ||
+			devMode != oidcConfig.DevMode ||
+			accessTokenType != oidcConfig.AccessTokenType.String() ||
+			accessTokenRoleAssertion != oidcConfig.AccessTokenRoleAssertion ||
+			clockSkew.String() != oidcConfig.ClockSkew.String() ||
+			!reflect.DeepEqual(additionalOrigins, oidcConfig.GetAdditionalOrigins()) {
```

Comment:
Why do we also need to compare to the system state? Isn't the comparision to the terraform state enough?